envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/EnvoyClustersFactory.kt

```diff
@@ -141,6 +146,25 @@ internal class EnvoyClustersFactory(
         return cluster.build()
     }
 
+    private fun mapPropertiesToThresholds(): List<CircuitBreakers.Thresholds> {
+        return properties.egress.commonHttp.circuitBreakers.thresholds.map {
+            val thresholdsBuilder = CircuitBreakers.Thresholds.newBuilder()
+            thresholdsBuilder.maxConnections = UInt32Value.of(it.maxConnections)
+            thresholdsBuilder.maxPendingRequests = UInt32Value.of(it.maxPendingRequests)
+            thresholdsBuilder.maxRequests = UInt32Value.of(it.maxRequests)
+            thresholdsBuilder.maxRetries = UInt32Value.of(it.maxRetries)
+            val priority = it.priority.toUpperCase()
+            if (priority == "DEFAULT") {
+                thresholdsBuilder.priority = RoutingPriority.DEFAULT
+            } else if (priority == "HIGH") {
+                thresholdsBuilder.priority = RoutingPriority.HIGH
+            } else {
+                thresholdsBuilder.priority = RoutingPriority.UNRECOGNIZED
```

Comment:
If this happens EC won't start. Is this ok? I think it is since blue/green deployments are a norm.